### PR TITLE
Add variable size volumes and support monitoring by default

### PR DESCRIPTION
**Pull request by Alex-Welsh**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -55,12 +55,37 @@ variable "multinode_vm_subnet" {
...
+
+variable "storage_disk_size" {
+  description = "Block storage root disk size for storage nodes in GB. Set to 0 on baremetal to use physical storage."
+  type = number
+  default = 100
 }
+
+variable "enable_monitoring" {
+  type = bool
+  default = true
```
Comment (jackhodgkiss) [inline variables.tf:90]: Could benefit from a description.

Comment (Alex-Welsh) [reply variables.tf:90]: I'm tempted to just remove the variable entirely and simply enable it. I just don't know if there is a downside to having the group declaration there and not using it

**Comment by Alex-Welsh** [comment]
> Quite like the approach of using the disk size to control machine type.

It doesn't exactly control the machine type, you still need to choose the flavour. It just only creates the block device if the size is >0

File: variables.tf (hunk 1 above)
Comment (Alex-Welsh) [reply variables.tf]: I've gone ahead and just changed the group declaration permanently. The monitoring functionality can be handled easily enough by just enabling/disabling services in the kayobe configuration

File: variables.tf [hunk 2]
```diff
@@ -55,12 +55,32 @@ variable "multinode_vm_subnet" {
   type = string
 }
 
-variable "compute_is_vm" {
-  description = "Should be set to true if VMs are being used for compute nodes"
-  type = bool
+variable "compute_disk_size" {
+  description = "Block storage root disk size for compute nodes in GB. Set to 0 on baremetal to use physical storage."
+  type = number
+  default = 100
```
Comment (markgoddard) [inline variables.tf:61]: This will use a disk by default, preventing baremetal from working is that what we want?

**Review by markgoddard — CHANGES_REQUESTED** [review]
README needs an update to cover how to use this.

File: variables.tf (hunk 2 above)
Comment (markgoddard) [reply variables.tf:61]: Most clouds don't provide bare metal, so we need to support VMs for portability. On SMS lab we'll probably mostly stick with bare metal as it's more realistic and there's plenty available

Comment (Alex-Welsh) [reply variables.tf]: I've removed the defaults but updated the README to properly explain what's going on

**Pull request merged**